rfd/0057-automatic-user-provisioning.md

```diff
@@ -134,11 +134,115 @@ Groups will not be cleaned up and will be created once and be reused
 this is to avoid files created with specified groups will remain
 accessible between sessions to users in those groups.
 
+
 ### Multiple matching roles
 
 Automatic user provisioning will require that all roles matching a
 node via `labels` have `create_host_user=true`
 
+
+## Update: allow host users to not be deleted after session end
+
+This will deprecate the current `create_host_user` option and replace
+it with an option called `create_host_user_mode`. The new option will
+have 3 possible settings:
+- `off`: disables host user creation
+- `drop`: deletes users after the session ends (current behaviour)
+- `remain`: leaves host users after the session ends.
+
+If the deprecated `create_host_user` option is specified and
+`create_host_user_mode` is not, it will default to using `drop` when
+it is true, and `off` when it is false.
+
+If `create_host_user_mode` is set, its setting will always be used
+instead of the `create_host_user` setting.
+
+If multiple roles matching a node specify the `create_host_user_mode`
+option with both `drop` and `remain`, the more restrictive `drop`
+option will be used.
```

Comment:
What is the downside to letting `remain` take precedence over `drop`?

Reply:
It should be a role option, yeah. I think it's ok to default to `keep`, then it will also be consistent with automatic user provisioning for database and Windows access which always keep the user. As long as there's no surprise change in behavior for existing users which are using the current setting.

That said, I think that the option on the SSH service should be updated accordingly as well, and converted to an enum: `create_host_user_mode: off|keep|drop`.

Reply:
@lxea Yeah I think you're right, it might get confusing. Let's not change the service option for now then.

Reply:
When implementing this we should give users the flexibility to set both `create_host_user` (for components that don't understand the new option) and `create_host_user_mode`, because if we blindly set `create_host_user` to true whenever `create_host_user_mode` is not off/empty we'll always end up with the `drop` behavior on old nodes, which may not be what the role actually intends.

Reply:
@espadolini so essentially you mean if `create_host_user=off && create_host_user_mode=remain`, when connecting to an old node, the node should read it as `create_host_user=off`, right? 

Is there any real benefit to this flexibility? I think having old nodes default to drop behavior when `create_host_user_mode=remain` is reasonable.

If we want to disable host user creation on a specific node, including old nodes, we can allow the user to set `create_host_user=off` or `create_host_user_mode=off` and prioritize that setting over role options.

Reply:
I think we should deprecate existing "create_host_user" option so we don't have to maintain both and follow our normal deprecation logic:

- If "create_host_user_mode" is set, it always takes precedence.
- Keep "create_host_user" for backwards compatibility and respect it if it's set (unless there's "creaet_host_user_mode") but remove it from all user-facing docs.

@lxea Can you update the RFD with this deprecation plan?

Reply:
I agree with the decision here, lgtm

Reply:
I think the service-level "mode" option is not needed as it may make things confusing around conflict resolution as we discussed above. Just keeping the "on/off" toggle we have should be fine for now IMO.